lib/ecto/adapters/exqlite.ex

```diff
@@ -299,4 +299,36 @@ defmodule Ecto.Adapters.Exqlite do
       :ok = Exqlite.Sqlite3.close(db)
     end
   end
+
+  defp dump_versions(config) do
+    table = config[:migration_source] || "schema_migrations"
+
+    # `.dump` command also returns CREATE TABLE which will clash with CREATE we already run in dump_schema
+    # So we set mode to insert which makes every SELECT statement to issue the result
+    # as the INSERT statements instead of pure text data.
+    case run_with_cmd("sqlite3", [
+           config[:database],
+           ".mode insert #{table}",
+           "SELECT * FROM #{table}"
+         ]) do
+      {output, 0} -> {:ok, output}
+      {output, _} -> {:error, output}
+    end
+  end
+
+  defp dump_schema(config) do
+    case run_with_cmd("sqlite3", [config[:database], ".schema"]) do
+      {output, 0} -> {:ok, output}
+      {output, _} -> {:error, output}
+    end
+  end
+
+  defp run_with_cmd(cmd, args) do
+    unless System.find_executable(cmd) do
```

Comment:
The sqlite3 release comes with a cli tool. I wonder if we should just package it as well with the binary.

Reply:
@warmwaffles I'm on the fence about this :) I can se both pros and cons

Reply:
Well, if we require them to have an sqlite binary on their machine, I would assume most distros come with their own sqlite3-dev package that we could require them to have. On the other hand, I like this being a single standalone package.

Reply:
@warmwaffles I just realised that we build sqlite on install anyway. So yeah, why not include the binary